pkg/connector/destination.go

```diff
@@ -248,13 +279,37 @@ func (d *Destination) preparePluginCall() (func(), error) {
 
 func (d *Destination) configure(ctx context.Context) error {
 	d.Instance.logger.Trace(ctx).Msg("configuring destination connector plugin")
-	err := d.plugin.Configure(ctx, d.Instance.Config.Settings)
+	_, err := d.plugin.Configure(ctx, pconnector.DestinationConfigureRequest{Config: d.Instance.Config.Settings})
 	if err != nil {
 		return cerrors.Errorf("could not configure destination connector plugin: %w", err)
 	}
 	return nil
 }
 
+func (d *Destination) open(ctx context.Context) error {
+	d.Instance.logger.Trace(ctx).Msg("opening destination connector plugin")
+	_, err := d.plugin.Open(ctx, pconnector.DestinationOpenRequest{})
+	if err != nil {
+		return cerrors.Errorf("could not open destination connector plugin: %w", err)
+	}
+	return nil
+}
+
+func (d *Destination) run(ctx context.Context) error {
+	d.Instance.logger.Trace(ctx).Msg("running destination connector plugin")
+	ctx, d.stopStream = context.WithCancel(ctx)
+	stream := d.plugin.NewStream()
+	err := d.plugin.Run(ctx, stream)
```

Comment:
Can the `Run` method return the stream?

Reply:
That would be great 😄 Although then we would need that abstraction layer (the one we removed now). The idea behind this change is that the interface is _exactly_ the same as what the connector implements (well, the connector SDK), so it's hopefully simpler to reason about. The connector needs to get a stream in `Run`, while Conduit needs a stream after it calles `Run`. This is now the middle ground, Conduit creates the stream and gives it to the connector, then both need to fetch the right "side" of the stream (conduit gets the client, the connector gets the server).